tests/scripts/thread-cert/Cert_7_1_02_BorderRouterAsRouter.py

```diff
@@ -107,6 +114,53 @@ def test(self):
             if addr[0:10] == '2001:2:0:1' or addr[0:10] == '2001:2:0:2':
                 self.assertTrue(self.nodes[LEADER].ping(addr))
 
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER = pv.vars['ROUTER']
+        MED = pv.vars['MED']
+        SED = pv.vars['SED']
+        _rpkts = pkts.filter_wpan_src64(ROUTER)
+
+        # Step 2: The DUT MUST send properly formatted MLE Advertisements
+        # The DUT MUST send a CoAP Server Data Notification message
+        # with the server’s information (Prefix, Border Router) to the Leader
+        _rpkts.filter_mle_cmd(MLE_CHILD_ID_REQUEST).must_next()
+        _rpkts.filter_mle_cmd(MLE_ADVERTISEMENT).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, ROUTE64_TLV} == set(p.mle.tlv.type))
+        _rpkts.filter_coap_request(SVR_DATA_URI).must_next()
+
+        # Step 3: Automatically transmits a 2.04 Changed CoAP response to the DUT
+        # Step 4: Router_1 MUST attempt to reattach to its original partition by
+
+        _rpkts.filter_mle_cmd(MLE_DATA_RESPONSE).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV} == set(p.mle.tlv.
+                                                                                                           type))
+
+        # Step 5: MED_1 to attach to DUT and request complete network data
+        # Step 6: The DUT MUST send an MLE Child ID Response to MED_1
+        p = _rpkts.filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next()
+        p.must_verify(lambda p: p.wpan.dst64 == MED and {Ipv6Addr('2001:2:0:1::'),
+                                                         Ipv6Addr('2001:2:0:2::')} == set(p.thread_nwd.tlv.prefix))
+        _rpkts_med = _rpkts.copy()
+
+        # Step 7: SED_1 to attach to DUT and request only stable data
+        # Step 8: The DUT MUST send an MLE Child Response to SED_1
+        _rpkts.filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next().must_verify(
+            lambda p: p.wpan.dst64 == SED and {Ipv6Addr('2001:2:0:1::')} == set(p.thread_nwd.tlv.prefix))
+        _rpkts_sed = _rpkts.copy()
+
+        # Step 9: SED_1 and MED_1 send its configured global address to the DUT
+        # Step 10: The DUT MUST send a Child Update Response to MED_1 & SED_1
+        _rpkts_med.filter_mle_cmd(MLE_CHILD_UPDATE_RESPONSE).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, MODE_TLV, ADDRESS_REGISTRATION_TLV} <= set(p.mle.tlv.type
+                                                                                     ) and p.wpan.dst64 == MED)
+        _rpkts_sed.filter_mle_cmd(MLE_CHILD_UPDATE_RESPONSE).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, MODE_TLV, ADDRESS_REGISTRATION_TLV} <= set(p.mle.tlv.type
```

Comment:
Can we verify the global addresses in these steps?

Reply:
already verify extaddrs

Reply:
Actually I meant the configured global addresses with prefix 2001:2:0:1:: or 2001:2:0:2::

Reply:
Referring to the test plan, only needs to check the 3 tlvs and the registration addresses configured by the Child in step
9